File: 2018/pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        int key = atoi(argv[1]) % 26;
        if (key >= 0)
        {
            string ptext = get_string("plaintext:  ");
            printf("ciphertext: ");

            for (int i = 0, n = strlen(ptext); i < n; i++)
            {
                if (isalpha(ptext[i]))
                {
                    if (isupper(ptext[i]))
                    {
                        printf("%c", (ptext[i] + key > 90 ? ptext[i] + key - 26 : ptext[i] + key));
                    }
                    else
                    {
                        printf("%c", (ptext[i] + key > 122 ? ptext[i] + key - 26 : ptext[i] + key));
                    }
                }
                else
                {
                    printf("%c", ptext[i]);
                }
            }
            printf("\n");
        }
        else
        {
            printf("Key must be a non-negative integer.\n");
        }
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}

File: 2018/pset6/caesar/caesar.py
from cs50 import get_string
import sys

key = int(sys.argv[1]) % 26
if key > 0:
    pt = get_string("plaintext: ")
    print("ciphertext: ", end='')
    for c in pt:
        if c.isalpha():
            if c.isupper():
                print(chr(ord(c) + key - 26) if ord(c) + key > 90 else chr(ord(c) + key), end='')
            else:
                print(chr(ord(c) + key - 26) if ord(c) + key > 122 else chr(ord(c) + key), end='')
        else:
            print(c, end='')
    print("")
else:
    print("Key must be a non-negative integer.")

